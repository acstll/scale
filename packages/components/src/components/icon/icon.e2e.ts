import { newE2EPage } from '@stencil/core/testing';

describe('t-text', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-text>this is a tag</t-text>');
    const element = await page.find('t-text');
    expect(element).toHaveClass('hydrated');
  });
});
