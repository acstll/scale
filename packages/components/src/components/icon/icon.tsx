import { Component, Prop, h, Host } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import jss from 'jss';
import preset from 'jss-preset-default';

jss.setup(preset());

const styles = {
  icon: {
    background: 'red',
  },
};

@Component({
  tag: 't-icon',
  styleUrl: 'icon.css',
})
export class Icon {
  /** (optional) Tag class */
  @Prop() public customClass?: string = '';
  @Prop() public name: string;
  @Prop() public path: string;

  stylesheet: any;

  componentWillLoad() {
    this.stylesheet = jss.createStyleSheet(styles);
  }

  public render() {
    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <svg
          class={this.getCssClassMap()}
          width="24"
          height="24"
          viewBox="0 0 26 26"
        >
          <path d={this.path} stroke="black" fill="transparent" />
        </svg>
      </Host>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      this.stylesheet.classes.icon,
      this.name && this.name,
      this.customClass && this.customClass
    );
  }
}
