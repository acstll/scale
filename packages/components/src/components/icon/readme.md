# t-icon



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description          | Type     | Default     |
| ------------- | -------------- | -------------------- | -------- | ----------- |
| `customClass` | `custom-class` | (optional) Tag class | `string` | `''`        |
| `name`        | `name`         |                      | `string` | `undefined` |
| `path`        | `path`         |                      | `string` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
