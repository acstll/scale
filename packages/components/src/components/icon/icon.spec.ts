import { newSpecPage } from '@stencil/core/testing';
import { Icon } from './icon';
import jss from 'jss';

describe('Icon', () => {
  let element;
  beforeEach(async () => {
    element = new Icon();
    element.stylesheet = jss.createStyleSheet({
      icon: {
        background: 'red',
      },
    });
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Icon],
      html: `<t-icon path="d 10 10">Label</t-icon>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle a custom css class', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');
  });

  it('should have a default css class', () => {
    expect(element.getCssClassMap()).toContain('icon');
  });
});
