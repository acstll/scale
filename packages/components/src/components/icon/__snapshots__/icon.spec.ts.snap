// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot 1`] = `
<t-icon path="d 10 10">
  Label
  <style>
    .icon-0-0-2 {
  background: red;
}
  </style>
  <svg class="icon-0-0-2" height="24" viewBox="0 0 26 26" width="24">
    <path d="d 10 10" fill="transparent" stroke="black"></path>
  </svg>
</t-icon>
`;
